plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.tsx

```diff
@@ -44,26 +43,14 @@ class ServiceConnectionEndpointList extends React.Component {
     return getDisplayValue(protocol);
   }
 
-  getHostPortValue(portDefinition, service) {
-    if (!service.requirePorts) {
-      return <Trans render="span">Auto Assigned</Trans>;
-    }
-
-    let hostPortValue = portDefinition.hostPort;
-
-    if (!hostPortValue) {
-      hostPortValue = portDefinition.port;
```

Comment:
@pierrebeitz This is missing in your version

Reply:
we can't get there, because we'll return early in line 47 if we don't have a `hostPort`, right?

Reply:
i'm assuming the state of 2b8cc9cedef478bd71949e8e54cf9523fe844533: 

```typescript
  getHostPortValue(portDefinition) {
    if (!portDefinition.hostPort) {                          <-- early return if no hostPort
      return <Trans render="span">Auto Assigned</Trans>;
    }

    let hostPortValue = portDefinition.hostPort;

    if (!hostPortValue) {                                    <-- this checks for "no host port?". can't be, because we already returned above.
      hostPortValue = portDefinition.port;
    }

    if (hostPortValue) {                                     <-- this returns if a hostPort is set
      return (
        <EndpointClipboardTrigger
          command={getDisplayValue(hostPortValue.toString())}
        />
      );
    }

    return getDisplayValue(hostPortValue);                     <-- we can't get here, because we returned when hostPort is set and when it was not set.
  }
```

with the above findings:

```diff
  getHostPortValue(portDefinition) {
    if (!portDefinition.hostPort) {
      return <Trans render="span">Auto Assigned</Trans>;
    }

    let hostPortValue = portDefinition.hostPort;

-    if (!hostPortValue) {
-      hostPortValue = portDefinition.port;
-    }

-    if (hostPortValue) {
      return (
        <EndpointClipboardTrigger
          command={getDisplayValue(hostPortValue.toString())}
        />
      );
-    }

-    return getDisplayValue(hostPortValue);
  }
```

then 

```diff
  getHostPortValue(portDefinition) {
    if (!portDefinition.hostPort) {
      return <Trans render="span">Auto Assigned</Trans>;
    }

    let hostPortValue = portDefinition.hostPort;             <-- this can now be a const. or we just inline `hostPort`

    return (
      <EndpointClipboardTrigger
        command={getDisplayValue(hostPortValue.toString())}
      />
    );
  }
```

so

```diff
  getHostPortValue(portDefinition) {
    if (!portDefinition.hostPort) {
      return <Trans render="span">Auto Assigned</Trans>;
    }

-    let hostPortValue = portDefinition.hostPort;

    return (
      <EndpointClipboardTrigger
-        command={getDisplayValue(hostPortValue.toString())}
+        command={getDisplayValue(portDefinition.hostPort.toString())}
      />
    );
  }
```

then 

```diff
  getHostPortValue(portDefinition) {
    if (!portDefinition.hostPort) {
      return <Trans render="span">Auto Assigned</Trans>;
    }

    return (
      <EndpointClipboardTrigger
        command={getDisplayValue(portDefinition.hostPort.toString())}
      />
    );
  }
```

and then some syntactic trickery brings us to the proposed version. WDYT?

Reply:
OK, got it